cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/MultiThreadDatasetCache.java

```diff
@@ -80,8 +82,12 @@ public void onRemoval(RemovalNotification<Thread, DynamicDatasetCache> notificat
           @Override
           @ParametersAreNonnullByDefault
           public SingleThreadDatasetCache load(Thread thread) throws Exception {
-            return new SingleThreadDatasetCache(
+            SingleThreadDatasetCache cache = new SingleThreadDatasetCache(
               instantiator, txClient, namespace, runtimeArguments, metricsContext, staticDatasets);
+            for (MultiThreadTransactionAware<?> txAware : multiThreadTxAwares) {
+              cache.addExtraTransactionAware(txAware);
```

Comment:
Yes. The `MultiThreadTransactionAware` will create different instance for different thread, pretty much like how the `MultiThreadDatasetCache` operates.